src/main/kotlin/org/serenityos/jakt/folding/JaktFoldingBuilder.kt

```diff
@@ -0,0 +1,51 @@
+package org.serenityos.jakt.folding
+
+import com.intellij.lang.ASTNode
+import com.intellij.lang.folding.FoldingBuilderEx
+import com.intellij.lang.folding.FoldingDescriptor
+import com.intellij.openapi.editor.Document
+import com.intellij.openapi.editor.FoldingGroup
+import com.intellij.openapi.project.DumbAware
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+import com.intellij.refactoring.suggested.endOffset
+import com.intellij.refactoring.suggested.startOffset
+import org.serenityos.jakt.psi.declaration.JaktImportMixin
+import org.serenityos.jakt.psi.findChildrenOfType
+
+class JaktFoldingBuilder : FoldingBuilderEx() {
+
+    override fun buildFoldRegions(root: PsiElement, document: Document, quick: Boolean): Array<FoldingDescriptor> {
+        val descriptors = arrayListOf<FoldingDescriptor>()
+
+        // Fold import lists
+        val imports = root.findChildrenOfType<JaktImportMixin>()
+        for (import in imports) {
+            val importBraceList = import.importBraceList
+            if (importBraceList != null) {
+                descriptors.add(FoldingDescriptor(import, importBraceList.textRange))
+            }
+        }
+
+        return descriptors.toTypedArray()
+    }
+
+    override fun getPlaceholderText(node: ASTNode): String? {
+        if (node.psi is JaktImportMixin) {
+            val import = node.psi as JaktImportMixin
+            val importCount = import.importBraceList?.importBraceEntryList?.size ?: return null
+            return "{ <$importCount items> }"
+        }
+        return null
+    }
+
+    override fun isCollapsedByDefault(node: ASTNode): Boolean {
+        if (node.psi is JaktImportMixin) {
+            val import = node.psi as JaktImportMixin
+            // Automatically fold any imports with more than 1 item.
+            // FIXME: Fold based on whether it spans multiple lines.
```

Comment:
This seems at first glance like it should be doable if you want to quickly add it, something like `if (import.importBraceList?.text?.includes('\n') == true)`